components/feature/accounts-push/src/main/java/mozilla/components/feature/accounts/push/FxaPushSupportFeature.kt

```diff
@@ -58,26 +76,40 @@ class FxaPushSupportFeature(
 internal class AccountObserver(
     private val context: Context,
     private val accountManager: FxaAccountManager,
-    private val observer: DeviceConstellationObserver,
+    private val push: AutoPushFeature,
     private val lifecycleOwner: LifecycleOwner,
     private val autoPause: Boolean
 ) : SyncAccountObserver {
+
     private val logger = Logger("AccountObserver")
+    private val constellationObserver = ConstellationObserver(context, push)
 
     override fun onAuthenticated(account: OAuthAccount, authType: AuthType) {
+        // We need a new subscription only when we have a new account.
+        // This is removed when an account logs out.
+        if (authType != AuthType.Existing) {
```

Comment:
What about `Recovered`? In that case we probably have a valid subscription already, right?

Reply:
I'm not really sure what changes fall behind the `Recovered` state from FxA's expectations, so I'll go with your recommendation.